source/slang/slang-emit-cpp.cpp

```diff
@@ -2589,34 +2589,43 @@ void CPPSourceEmitter::emitModuleImpl(IRModule* module)
+                                //DiagnosticSink sink(linkage->getSourceManager());
                                 CompileResult result =
-                                    targetProgram->getOrCreateEntryPointResult(index, &sink);
+                                    targetProgram->getOrCreateEntryPointResult(index, sink);
 
                                 Slang::ComPtr<ISlangBlob> blob;
                                 if (SLANG_FAILED(result.getBlob(blob)))
                                 {
-                                    SLANG_UNEXPECTED("No blob to emit");
-                                    return;
+                                    StringBuilder builder;
+                                    sink->diagnoseRaw(Severity::Error, UnownedStringSlice
+                                        ("Slang heterogeneous error: No blob to emit\n"));
+                                    m_writer->emit("size_t __");
+                                    m_writer->emit(entryPointName);
+                                    m_writer->emit("Size = 0;\n");
+                                    m_writer->emit("unsigned char __");
+                                    m_writer->emit(entryPointName);
+                                    m_writer->emit("[];\n");
                                 }
 
-                                auto ptr = (const unsigned char*)blob->getBufferPointer();
-
-                                m_writer->emit("size_t __");
-                                m_writer->emit(entryPointName );
-                                m_writer->emit("Size = ");
-                                m_writer->emitInt64(blob->getBufferSize());
-                                m_writer->emit(";\n");
-
-                                m_writer->emit("unsigned char __");
-                                m_writer->emit(entryPointName );
-                                m_writer->emit("[] = {");
-                                for (unsigned int i = 0; i < blob->getBufferSize() - 1; i++) {
-                                    m_writer->emitUInt64(ptr[i]);
-                                    m_writer->emit(", ");
+                                else {
+                                    auto ptr = (const unsigned char*)blob->getBufferPointer();
+
+                                    m_writer->emit("size_t __");
+                                    m_writer->emit(entryPointName);
+                                    m_writer->emit("Size = ");
+                                    m_writer->emitInt64(blob->getBufferSize());
+                                    m_writer->emit(";\n");
+
+                                    m_writer->emit("unsigned char __");
+                                    m_writer->emit(entryPointName);
+                                    m_writer->emit("[] = {");
+                                    for (unsigned int i = 0; i < blob->getBufferSize() - 1; i++) {
+                                        m_writer->emitUInt64(ptr[i]);
```

Comment:
That here I don't think we want the -1? Oh okay maybe you have that so there isn't a hanging , at the end. You don't need to worry about this though as C/C++ allow this extra comma (just for these kinds of situations). 
Also note special handling means that if getBufferSize() == 0, would be a bug/crash. That the blob being 0 sized arguably doesn't make much sense here in practice but, the simpler loop with a , at the end would not have this issue.

```
// To avoid repeatedly calling getBufferSize in loop (the C++ compiler can't generally see into blob implementation, that the value will not change).
const size_t bufferSize = blob->getBufferSize();
for (size_t i = 0; i < bufferSize; ++i)
{
   // If I don't cast then it won't find the overload unambiguously (arguably we should have some explicit sized methods)
  // But I don't want to use emitUInt64 because that's likely more unnecessary work 
   m_writer->emit(int(ptr[i]));
}
```